pkg/odo/cli/component/create.go

```diff
@@ -567,70 +564,77 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 			if err != nil {
 				return err
 			}
-
 			return nil
 		}
+	}
 
-		if isDevfileRegistryPresent && !co.forceS2i {
-			// Categorize the sections
-			log.Info("Validation")
+	if isDevfileRegistryPresent {
+		// Categorize the sections
 
-			// Since we need to support both devfile and s2i, so we have to check if the component type is
-			// supported by devfile, if it is supported we return and will download the corresponding devfile later,
-			// if it is not supported we still need to run all the codes related with s2i after devfile compatibility check
+		// Since we need to support both devfile and s2i, so we have to check if the component type is
+		// supported by devfile, if it is supported we return and will download the corresponding devfile later,
+		// if it is not supported we still need to run all the codes related with s2i after devfile compatibility check
 
-			hasComponent := false
-			devfileExistSpinner := log.Spinner("Checking devfile existence")
+		hasComponent := false
+		var devfileExistSpinner *log.Status
```

Comment:
As someone who wrote the `log` library and functionality, I'd suggest we avoid declaring a spinner like this.. 

You may run into issues in the future where the spinner will encounter a nil pointer.